src/pydantic_zarr/v3.py

```diff
@@ -126,6 +176,45 @@ class ArraySpec(NodeSpec, Generic[TAttr]):
     storage_transformers: tuple[NamedConfig, ...]
     dimension_names: tuple[str | None, ...] | None  # todo: validate this against shape
 
+    def model_dump(
+        self,
+        *,
+        mode: Literal["json", "python"] | str = "python",  # noqa: PYI051
+        include: IncEx | None = None,  # type: ignore[override]
+        exclude: IncEx | None = None,  # type: ignore[override]
+        context: Any | None = None,
+        by_alias: bool | None = None,
+        exclude_unset: bool = False,
+        exclude_defaults: bool = False,
+        exclude_none: bool = False,
+        round_trip: bool = False,
+        warnings: (bool | Literal["none", "warn", "error"]) = True,
+        fallback: Callable[[Any], Any] | None = None,
+        serialize_as_any: bool = False,
+    ) -> dict[str, Any]:
+        """
+        Override this method because the Zarr V3 spec requires that the dimension_names
```

Comment:
It *feels* like there should be a nicer way to achieve this in `pydantic`, but looking through their docs I don't think there is :(